driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -382,23 +404,11 @@ private void retry(final boolean retryCurrent, ConsistencyLevel newConsistencyLe
             if (newConsistencyLevel != null)
                 this.retryConsistencyLevel = newConsistencyLevel;
 
-            // We should not retry on the current thread as this will be an IO thread.
-            manager.executor().execute(new Runnable() {
-                @Override
-                public void run() {
-                    if (queryStateRef.get().isCancelled())
-                        return;
-                    try {
-                        if (retryCurrent) {
-                            if (query(h))
-                                return;
-                        }
-                        sendRequest();
-                    } catch (Exception e) {
-                        setFinalException(null, new DriverInternalError("Unexpected exception while retrying query", e));
-                    }
-                }
-            });
+            if (queryStateRef.get().isCancelled())
+                return;
+
+            if (!retryCurrent || !query(h))
```

Comment:
Nice,  I remember seeing retries fill up the Cluster executor if your pool timeout > 0 on retries and pool was full.